services/steam/steam.service.js

```diff
@@ -0,0 +1,395 @@
+'use strict'
+
+const Joi = require('joi')
+const BaseJsonService = require('../base-json')
+const { metric, formatDate } = require('../../lib/text-formatters')
+const { age: ageColor } = require('../../lib/color-formatters')
+const prettyBytes = require('pretty-bytes')
+
+const steamCollectionSchema = Joi.object({
+  response: Joi.object()
+    .keys({
+      result: Joi.number()
+        .integer()
+        .min(1)
+        .max(1)
+        .required(),
+      resultcount: Joi.number()
+        .integer()
+        .min(0)
+        .max(1)
+        .required(),
+      collectiondetails: Joi.array()
+        .items(
+          Joi.object({
+            publishedfileid: Joi.string(),
+            result: Joi.number().integer(),
+            children: Joi.array(),
+          })
+        )
+        .required(),
+    })
+    .required(),
+}).required()
+
+const steamFileSchema = Joi.object({
+  response: Joi.object()
+    .keys({
+      result: Joi.number()
+        .integer()
+        .min(1)
+        .required(),
+      resultcount: Joi.number()
+        .integer()
+        .min(0)
+        .required(),
+      publishedfiledetails: Joi.array().items(
+        Joi.object({
+          publishedfileid: Joi.string().required(),
+          result: Joi.number()
+            .integer()
+            .required(),
+          file_size: Joi.number().integer(),
+          time_created: Joi.number().integer(),
+          subscriptions: Joi.number().integer(),
+          favorited: Joi.number().integer(),
+          lifetime_subscriptions: Joi.number().integer(),
+          lifetime_favorited: Joi.number().integer(),
+          views: Joi.number().integer(),
+        })
+      ),
+    })
+    .required(),
+}).required()
+
+class SteamCollectionFiles extends BaseJsonService {
+  async fetch({ collectionId }) {
+    const url =
+      'https://api.steampowered.com/ISteamRemoteStorage/GetCollectionDetails/v1/?format=json'
+    return this._requestJson({
+      url,
+      schema: steamCollectionSchema,
+      errorMessages: {
+        400: 'bad request',
+      },
+      options: {
+        method: 'POST',
+        form: {
+          collectioncount: '1',
+          'publishedfileids[0]': collectionId,
+        },
+      },
+    })
+  }
+
+  static render({ size }) {
+    return { message: metric(size), color: 'green' }
+  }
+
+  async handle({ collectionId }) {
+    const json = await this.fetch({ collectionId })
+    if (json.response.collectiondetails[0].result === 1) {
+      return this.constructor.render({
+        size: json.response.collectiondetails[0].children.length,
+      })
+    } else {
+      return { message: 'collection not found', color: 'red' }
```

Comment:
Instead of manually constructing error reponses, we can throw an exception, for example:

https://github.com/badges/shields/blob/302c8606ff4c975b6310fe2cdfe58f92cede290b/services/npm/npm-version.service.js#L95-L97

This will ensure consistent behaviour and formatting with other errors. Also in the cases where we are failing with "file not found", "collection not found" etc can you add test cases to the test suite covering those cases. You can use `nock` to provide a mock response recreating these conditions. There are some tests using mocked responses in the NPM licence tests that you can use as examples: https://github.com/badges/shields/blob/master/services/npm/npm-license.tester.js